compiler/modules.nim

```diff
@@ -116,7 +116,7 @@ proc newModule(fileIdx: int32): PSym =
   result.kind = skModule
   let filename = fileIdx.toFullPath
   result.name = getIdent(splitFile(filename).name)
-  if not isNimIdentifier(result.name.s):
+  if result.name.s != "-" and not isNimIdentifier(result.name.s):
```

Comment:
If you fake the module name to be 'stdin', why do you need this check? Also the name 'stdin' will cause conflicts with 'system.stdin'.